experimental/dds/tree2/src/feature-libraries/untypedTree.ts

```diff
@@ -298,4 +298,18 @@ export interface EditableTreeEvents {
 	 * @returns a visitor to traverse the subtree or `void`.
 	 */
 	subtreeChanging(upPath: UpPath): PathVisitor | void;
+
+	/**
+	 * Raised on a node right before a change is applied to one of its fields or the fields of a descendant node.
+	 * @param upPath - The path (upward) corresponding to the location of the node raising the event. Note that this is not
+	 * necessarily the node that is about to change; the change could be in a descendant of the node the `upPath` points to.
+	 */
+	beforeChange(upPath: UpPath): void;
```

Comment:
`beforeChange` and `afterChange` are [specced](https://microsoft-my.sharepoint-df.com/:w:/p/nsimons/EeiyFIgzDe1DnVKI33LkWb4BSFqQa6WcxAWapzswxPn-Iw?e=u1nz94) to pass an "args" object as the sole parameter. Right now the spec only has "target" on that args object, so we don't need the path.

Reply:
The target in this case would be the actual tree node, right? So if I wanted to validate it in tests, I'd use a cursor to navigate to the node that I'm expecting will be passed to the listener as `args.target`, and reference-equals it to the actual `args.target`?

Reply:
Sorry, I didn't see your reply until just now. The type of `target` should match the type of the object that the event was registered on, so yes, a TreeNode. The type will be changing soon, but that should be okay for now. I think you could just hold on to the node that you registered the event on in the test, and reference equal check against that.